src/graph/graphQL.ts

```diff
@@ -72,15 +77,34 @@ export function prettyPrintErrors(errors: GraphQLError[], query?: string): strin
 /**
  * Resolve and read the contents of a GrapqQL query or subscription file
  * @param {string} path
+ * @param {string} current
+ * @param {{[p: string]: string | boolean | number}} parameters
  * @returns {string}
  */
-export function resolveAndReadFileSync(path: string, current: string = appRoot.path): string {
+export function resolveAndReadFileSync(path: string,
+                                       current: string = appRoot.path,
+                                       parameters: {
+                                           [name: string]: string | boolean | number;
+                                       }): string {
     if (!path.endsWith(".graphql")) {
         path = `${path}.graphql`;
     }
     const absolutePath = p.resolve(current, path);
     if (fs.existsSync(absolutePath)) {
-        return fs.readFileSync(absolutePath).toString();
+        let content = fs.readFileSync(absolutePath).toString();
+        for (const key in parameters) {
+            if (parameters.hasOwnProperty(key)) {
+                const value: any = parameters[key];
+                if (typeof value === "string") {
+                    content = content.replace(`$${key}`, `"${value}"`);
+                } else if (typeof value === "number") {
+                    content = content.replace(`$${key}`, `${value}`);
+                } else if (typeof value === "boolean") {
```

Comment:
Seems the "number" and "boolean" cases could be combined into one. Or you could get rid of the conditional entirely with something like:

```typescript
content = content.replace(new RegExp(`\\$${key}\\b`, "g"), JSON.stringify(parameters[key]));
```

Reply:
Thanks, that's good.